sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/WebPubSubConnectionContext.cs

```diff
@@ -78,9 +89,38 @@ public class WebPubSubConnectionContext
         /// <param name="userId">User Id.</param>
         /// <param name="signature">Signature of the connection.</param>
         /// <param name="origin">Origin of the event.</param>
-        /// <param name="states">Connection states. Value in <see cref="JsonElement"/> type. Use the <see cref="TryGetState{T}(string, out T)"/> method to tey getting the vaule deserialized as a specific type.</param>
+        /// <param name="states">Connection states. Value in <see cref="BinaryData"/> type.</param>
         /// <param name="headers">Connection request headers.</param>
-        public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName, string hub, string connectionId, string userId = null, string signature = null, string origin = null, IReadOnlyDictionary<string, object> states = null, IReadOnlyDictionary<string, string[]> headers = null)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName, string hub, string connectionId, string userId, string signature, string origin, IReadOnlyDictionary<string, object> states, IReadOnlyDictionary<string, string[]> headers)
+            : this(
+                eventType,
+                eventName,
+                hub,
+                connectionId,
+                userId,
+                signature,
+                origin,
+                states?.ToDictionary(
+                    p => p.Key,
+                    p => p.Value as BinaryData ?? BinaryData.FromObjectAsJson(p.Value)),
```

Comment:
If an older version called this ctor, `p.Value` is actually the value of `JsonElement.ToString()`. `string` values and object values behaves differently when `JsonElement.ToString()`, for a string `A`, `JsonElement.ToString()` is `A` without quote.